source/tokens/TokenGenerator.hx

```diff
@@ -0,0 +1,79 @@
+package tokens;
+
+import flixel.FlxBasic;
+import flixel.FlxG;
+import flixel.group.FlxGroup;
+import flixel.util.FlxTimer;
+import tokens.Token;
+
+@:generic
+class TokenGenerator<T:Token> extends FlxBasic
+{
```

Comment:
If you make token inherit from Obstacle, then I don't think you need this, right? You could just use an ObjectGenerator to make tokens on a timer, just like how we use ObjectGenerators to make DeadlyObstacles